pkg/clusterconditions/clusterconditions.go

```diff
@@ -0,0 +1,83 @@
+// Package clusterconditions implements cluster conditions for
+// identifying metching clusters.
+//
+// https://github.com/openshift/enhancements/blob/master/enhancements/update/targeted-update-edge-blocking.md#cluster-condition-type-registry
+package clusterconditions
+
+import (
+	"context"
+	"fmt"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
+)
+
+type Condition interface {
+	// Valid returns an error if the condition is expected to be
+	// rejected by the Kubernetes API server.  For example, for
+	// missing or invalid data.
+	Valid(ctx context.Context, condition *configv1.ClusterCondition) error
+
+	// Match returns whether the condition matches the current
+	// cluster (true), does not match the current cluster (false),
+	// or fails to evaluate (error).
+	Match(ctx context.Context, condition *configv1.ClusterCondition) (bool, error)
+}
+
+// Registry is a registry of implemented condition types.
+var Registry map[string]Condition
+
+// Register registers a condition type, and panics on any name collisions.
+func Register(conditionType string, condition Condition) {
+	if existing, ok := Registry[conditionType]; ok && condition != existing {
+		panic(fmt.Sprintf("cluster condition %q already registered", conditionType))
+	}
+	if Registry == nil {
+		Registry = make(map[string]Condition, 1)
+	}
```

Comment:
Move before use above or why not just init in declaration.

Reply:
Fixed with 86b9ad96 -> b5f1d1df.  Not initializing in an `init()` here, because `Registry` is public, and crazy folks might set it back to `nil` after importing the module.